.travis.yml

```diff
@@ -5,8 +5,24 @@ env:
     - secure: "CfAXYEj34sBP3JuLguE9k6+0MXwVr924MrZ36dGMVp2Xvr2XvSyL4KEACHS9NalEl11wIOvrOmjfIj10KQI0G3SEsfDmSHPNYmkbtTQ0wrUVLFbis9PzKfCkmoDhUU0dZ0k3xMueDsHAkX6shKjJ5zxNhXwrarLgQE7TuyyBfLxnmekbkE5N8H1L38RLnF+2biyHql2SZggXwYCzRSLNQrqOWnG1XlT/N4wr/LvCxcpdeabs86AGAOUrMiISjm7oIWYrpAplPV1gtAkub0C5T1piCwaeJFcnHwXvg6pBFoSvf9Cw6mKkxrmPQ8QjH2RiZwWWJifIDSwYUCooQtYh5lqkr+HRWve6kgfbiSc99H/8i2LwV/jnJtFvrwC4jNmrt0haZH6KgJ3dTf3aOnpiCHV8gboSxZFMTJbdbc8XsNk0HYQgSysoRr0EVKs0iBqGSeqpFTHP6InroBsuYi9IDKMXGTIsJCatw0mtnMnacY2V8X8tmAoGj2bO+ueEI5hMFkdJRFx8YJoTVnv22dcHgYVghOJnF9gtOZxMaY7eUeNQFiU6ERC7LbC8TvkM+bK/VPQNyWsmyTxc2biyn22gOIXHGEyCO7I7u0v7lZew2dYVThxo73HDMyvoyxaCANHaPVV1bmgoGzX0tsJo4Y7eBo/tvkqS9a16MRoLvjDKVGk="
+    - DOCKER_USERNAME=joleesoftwire
+    - secure: "b+r9isyv2os6kGSFRKEViQaZJ5YfcT/TbInlFmvl5lyIfgnsWu5CR7F/Xqke+8m6Rp55ESTIVoYe+J/Ij9+UV64DxZhbq+03cyYCy4ItrIpyzG904ZsnuHQL2/1RX5QMRQ/zZtZG45OEOD/OSIspv8zzsjIY0YFV4wcs4ESRRycnGxSp0vP7xK/tkRAMi/9cab2/6U5hEd3/A8wf+GPA2R3IiIB5QcMZCWFgu0Qrn9MaD77+W/DxVFrw+0cc7yz755AF/SFyQPw0XTIMICCMhGa278bDBFaVvOML/AjEy+x2v57eWjMjm6y4nxNHEAksQoEAfhuE3LGPMn0txrJ+/+Upw6EJH+NfbFf0iUq+6H+d1NQij/gu9vND4SgYZZYwBjsehOfiuLmn6ice1h8lG5Ov1LvEWDieu7ay6jGWVdHPQhg7hVvYrcAjlfsI1UgFRCzVj7GjkttaI/u52OauX3t4fCAx3s6db1/9qINAflG33V8BnJNKKzPWTCTcbH0uf2OKk9naJMOflumTr0yEbMCvTok45AHUl/+cc+RQSyuJvY1fKOz/WpTVSZSv1t06kW5mVlf+qosknn0RJVvQjYpmXJ/XKK83jQZAicm9XOQQxZxnArRrqTNA/RNcpT+vgiU6dmwFSa/EyCB3vHk2MbzA4OwtSCehxZc68/9JBow="
+    - secure: "gsnQLulWFd+DHOzmprQfPyKaVNSh+/YVl4Q24j7AaqwmLtN5YX97MU59j0NcgxXEJijDqcrU+emNMoHew6viyp2zvgSfW/wxfK81dirKYS5wusFvf6OFtnzVMOWnDO54V5hQZq1sKbD6EtFtxSjjec9K5iq6gAxon1iKR7w0BZnZcRQoAKlcTJGrprzhkTxJq0BK38dW+H+1AQ4Znm5NNEW7XLdIdINPrVtaojfSP6FSxL5447kIF6xjOa1eblj6WXpf+vZei4Wox85nb9Li4Ql4nbth9GeIvhRt6aTVjZpFupBAngty1Jn5aLwYSxjmt1GzLvb+Sw2dekkX+oJBiUo1WmkxntISLNjSuXb3IHvjMaKSL7+HH3zgW4U4LsJoq1jLqY3B2CM26K3UpcTg40Rr6QuHOOiGUfhj1nu7YHTIlc1gzW0liCQddq7Ly0Axqx4MSylsHFisTGwNyzKYnDbHH6jBX3wdyfGRmoM4GZcT+PJKUp135qDFztD1eadAeE6hWjHQPWVoY/VZxma++DgELBmKfP6HlNBahlIsXVjI1icRteQ3LA/QDgPpFa9+q77jBJHeg3Cjg/Hz+L2EnFn9MUp5DIWITDQDRfBZQo1ghHfSQGoNEodh98otHL3szcpRckGG+Ej9mXDzFopQ5j0jojN4xtODvPI5EXBEL80="
 
-script:
+
+before-script:
 - docker build --target test --tag todoapp-test .
 - docker run todoapp-test src/tests
 - docker run -e TRELLO_TOKEN=$TRELLO_TOKEN -e TRELLO_KEY=$TRELLO_KEY todoapp-test src/tests_e2e
+- echo $DOCKER_PASSWORD | docker login -u $DOCKER_USERNAME --password-stdin
+
+script:
+- docker build --target production --tag joleesoftwire/todoapp:latest --tag joleesoftwire/todoapp:$TRAVIS_COMMIT .
+
+after-sucess:
+- echo $DOCKER_PASSWORD | docker login -u $DOCKER_USERNAME --password-stdin
+- docker login --username=joleesoftwire --password=$HEROKU_API_KEY registry.heroku.com
```

Comment:
Docker recommends using `--password-stdin`, like you did on the line above rather than `--password`. Either way it's a bit weird to do the two logins differently.